ConfigurationSystem/Client/CSShellCLI.py

```diff
@@ -0,0 +1,211 @@
+"""CSShellCmd class emulates the behaviour of a shell to edit the CS config.
+"""
+
+__RCSID__ = "$Id$"
+
+import cmd
+import os
+
+from DIRAC.Core.Base.CLI                                import CLI, colorize
+from DIRAC.ConfigurationSystem.Client.ConfigurationData import gConfigurationData
+from DIRAC.ConfigurationSystem.private.Modificator      import Modificator
+from DIRAC.Core.DISET.RPCClient                         import RPCClient
+
+class CSShellCLI( CLI ):
+
+  def __init__(self):
+    CLI.__init__(self)
+    self.serverURL   = ""
+    self.serverName  = ""
+    self.modificator = None
+    self.connected   = False
+    self.dirty       = False
+    self.root        = "/"
+
+    self.do_connect("")
+
+  def update_prompt(self):
+    if self.connected:
+      self.prompt = "[" + colorize(self.serverName, "green") + ":" + self.root + " ]% "
+    else:
+      self.prompt = "[" + colorize("disconnected", "red") + ":" + self.root + " ]% "
+
+  def do_connect(self, line):
+    """connect
+    Connect to the CS
+    Usage: connect <URL> (Connect to the CS at the specified URL)
+           connect       (Connect to the default CS URL of your config)
+    """
+    if line == "":
+      self.serverURL  = gConfigurationData.getMasterServer()
+      self.serverName = gConfigurationData.getName()
+    else:
+      self.serverURL  = self.serverName = line
+
+    if self.serverURL == None:
+      print "Unable to connect to the default server. Maybe you don't have a proxy ?"
+      return self.do_disconnect("")
+
+    print "Trying to connect to " + self.serverURL + "...",
+
+    self.modificator = Modificator(RPCClient(self.serverURL))
+    rv               = self.modificator.loadFromRemote()
+    rv2              = self.modificator.loadCredentials()
+
+    if rv['OK'] == False or rv2['OK'] == False:
```

Comment:
This is a patch release and I do not want to modify code that can affect logic. I am just moving code around